### PR TITLE
Delete transient permissions when peer dies

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
Not when the session is closed. See patches.

Closes: https://github.com/flatpak/xdg-desktop-portal/issues/1124

### CODE REVIEW

File: src/session.c [hunk 1]
```diff
@@ -270,6 +271,8 @@ close_sessions_in_thread_func (GTask *task,
       session_close (session, FALSE);
     }
```
Comment (jadahl) [inline src/session.c:261]: Perhaps this function (`close_sessions_in_thread_func()`) should get a new name now that it does more things than closing sessions. Alternatively, add a dedicated step in `peer_died_cb()` after the other two.

Comment (GeorgesStavracas) [reply src/session.c]: Went with the latter (the first version of this patch did exactly than before I moved it to `close_sessions_in_thread_func()`, it felt natural to move it back to `peer_died_cb()`)

File: src/xdg-desktop-portal.c [hunk 2]
```diff
@@ -229,6 +230,7 @@ peer_died_cb (const char *name)
 {
   close_requests_for_sender (name);
   close_sessions_for_sender (name);
+  xdp_session_persistence_delete_transient_permissions_for_sender (sender);
```
Comment (jadahl) [inline src/xdg-desktop-portal.c:233]: s/sender/name/

**Review by jadahl — CHANGES_REQUESTED** [review]

File: src/xdg-desktop-portal.c (hunk 2 above)
Comment (GeorgesStavracas) [reply src/xdg-desktop-portal.c]: Of course

**Pull request merged**